### PR TITLE
Add the Pimoroni Plasma 2040 board

### PR DESCRIPTION
**Pull request by jwillikers**
This PR adds the pin definitions, a couple constants for the current sensor, and a simple example.
The example just blinks the on-board RGB LED.

An example driving some DotStar LEDs would be a great future addition.
An example or functionality should also be added for measuring the current from the built-in current sensor.

### CODE REVIEW

File: boards/pimoroni-plasma-2040/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# [pimoroni-plasma-2040] - Board Support for the [Pimoroni Plasma 2040]
+
+You should include this crate if you are writing code that you want to run on
+a [Pimoroni Plasma 2040] - Swathe everything in rainbows with this all-in-one, USB-C powered controller for WS2812/Neopixel and APA102/Dotstar addressable LED strip.
+
+This crate includes the [rp2040-hal], but also configures each pin of the
+RP2040 chip according to how it is connected up on the Pimoroni Plasma 2040.
+
+[Pimoroni Plasma 2040]: https://shop.pimoroni.com/products/plasma-2040
+[rp-pico]: https://github.com/rp-rs/rp-hal/tree/main/boards/rp-pico
```
Comment (9names) [inline boards/pimoroni-plasma-2040/README.md:10]: Remove reference to pico here
```suggestion
```

File: boards/pimoroni-plasma-2040/examples/pimoroni_plasma_2040_blinky.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+#[entry]
+fn main() -> ! {
+    info!("Program start");
+    let mut pac = pac::Peripherals::take().unwrap();
+    let core = pac::CorePeripherals::take().unwrap();
+    let mut watchdog = Watchdog::new(pac.WATCHDOG);
+    let sio = Sio::new(pac.SIO);
+
+    // External high-speed crystal on the pico board is 12Mhz
```
Comment (9names) [inline boards/pimoroni-plasma-2040/examples/pimoroni_plasma_2040_blinky.rs:29]: Remove this comment. Board is not a pico, this isn't a HAL where user needs to care about crystal speed - that's what XOSC_CRYSTAL_FREQ is for
```suggestion
```

**Review by 9names — CHANGES_REQUESTED** [review]
Just a couple of nitpicks. Some of these changes should also happen in the Pico examples you copied from as well (you don't need to update the Pico examples as part of this PR if you don't want to)

File: boards/pimoroni-plasma-2040/README.md (hunk 1 above)
Comment (9names) [reply boards/pimoroni-plasma-2040/README.md:10]: Actually, update this path to match the git path for pimoroni-plasma-2040 instead.

Comment (9names) [reply boards/pimoroni-plasma-2040/README.md:10]:
```suggestion
[pimoroni-plasma-2040]: https://github.com/rp-rs/rp-hal/tree/main/boards/pimoroni-plasma-2040
```

Comment (jwillikers) [reply boards/pimoroni-plasma-2040/README.md]: That makes more sense. That snuck in there from the pimoroni-tiny2040's README which I copied. Fixed.

File: boards/pimoroni-plasma-2040/examples/pimoroni_plasma_2040_ws2812_led.rs [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        &mut pac.RESETS,
+    );
+
+    // Setup a delay for the LED blink signals:
+    let mut frame_delay =
+        cortex_m::delay::Delay::new(core.SYST, clocks.system_clock.freq().integer());
+
+    // Import the `sin` function for a smooth hue animation from the
+    // Pico rp2040 ROM:
+    let sin = pimoroni_plasma_2040::hal::rom_data::float_funcs::fsin::ptr();
```
[the remark this answers is not in the document]
Comment (jwillikers) [reply boards/pimoroni-plasma-2040/examples/pimoroni_plasma_2040_ws2812_led.rs]: I don't mind fixing in both places. Done.

File: boards/pimoroni-plasma-2040/examples/pimoroni_plasma_2040_blinky.rs (hunk 2 above)
Comment (jwillikers) [reply boards/pimoroni-plasma-2040/examples/pimoroni_plasma_2040_blinky.rs]: Removed for Plasma 2040 example and in the example from whence it came.

**Review by 9names — APPROVED** [review]
LGTM, thanks!

**Pull request merged**